Automattic/wp-calypso
# Theme Discovery: Add Partner collection carousel to Recommended view

Related to:
- https://github.com/Automattic/dotcom-forge/issues/4442

Builds on (merge is blocked by):
- https://github.com/Automattic/wp-calypso/pull/83799

After restoring the `Recommended` tab we need to add the Partner Collection Carousel inside the Theme's Showcase main view. The Collection Carousel must be in the third position and styled according to the Figma file available at the parent issue.

There is one thing that I need to fix, on mobile screens the Theme Showcase isn't working properly. I'll ask for some help on this PR.

## Proposed Changes

* Added the Partner Collection Carousel to the Theme List component as a child.
* With CSS we position it on the Nth row where n=3.
* Removed the link from the Partner collection definition.
* Removed unused code & cleaned up.

## Testing Instructions



* Open the Calypso Live Link
* Go to the LoTS and add `flags=themes/discovery-lots` 
* On the main view (Recommended tab) you should see that there is a Partner Collection Carousel.
* The Carousel must not have a "Learn more" link in the description.
* The Carousel must be fully functional.
* Click on see all, you must see the collection view of the partner themes.
* When clicking back, you should navigate to the recommended tab.

Note: The current PR does not include the changes from:
- https://github.com/Automattic/wp-calypso/pull/83803
So you will see that the Recommended tab shows 3 Theme Cards per row on non-collection Carousel components. After the aforementioned PR gets merged there will be more elements per row but this shouldn't affect the changes proposed in this PR.


Note 2: The current PR now includes the changes that show more than 3 Theme Cards per row.

Before:
![Captura de pantalla 2023-11-02 a las 14 32 27](https://github.com/Automattic/wp-calypso/assets/1989914/fc51361c-6891-4d26-b6c5-77b38035d80f)

After:
![Captura de pantalla 2023-11-02 a las 16 14 29](https://github.com/Automattic/wp-calypso/assets/1989914/8b040406-0162-402a-9ffa-f2cb93716f8b)


## Pre-merge Checklist
